pymc/tests/test_mixture.py

```diff
@@ -657,6 +658,103 @@ def test_iterable_single_component_warning(self):
         with pytest.warns(UserWarning, match="Single component will be treated as a mixture"):
             Mixture.dist(w=[0.5, 0.5], comp_dists=[Normal.dist(size=2)])
 
+    @pytest.mark.parametrize(
+        "weights, comp_dists, size, expected",
+        [
+            (
+                np.array([1, 0]),
+                [Normal.dist(-2, 5), Normal.dist(6, 3)],
+                None,
+                -2,
+            ),
+            (
+                np.array([0.4, 0.6]),
+                [Normal.dist(-2, 5), Normal.dist(6, 3)],
+                None,
+                2.8,
+            ),
+            (
+                np.array([0.5, 0.5]),
+                [Normal.dist(-2, 5), Exponential.dist(lam=1 / 3)],
+                None,
+                0.5,
+            ),
+            (
+                np.broadcast_to(np.array([0.4, 0.6]), (5, 3, 2)),
+                [Normal.dist(-2, 5), Normal.dist(6, 3)],
+                None,
+                np.full(shape=(5, 3), fill_value=2.8),
+            ),
+            (
+                np.array([0.4, 0.6]),
+                [Normal.dist(-2, 5, size=(5, 3)), Normal.dist(6, 3, size=(5, 3))],
+                None,
+                np.full(shape=(5, 3), fill_value=2.8),
+            ),
+            (
+                np.array([0.4, 0.6]),
+                [Normal.dist(-2, 5), Normal.dist(6, 3)],
+                (3,),
+                np.full(shape=(3,), fill_value=2.8),
+            ),
+            (
+                np.array([0.4, 0.6]),
+                [Normal.dist(-2, 5), Normal.dist(6, 3)],
+                (11, 7),
+                np.full(shape=(11, 7), fill_value=2.8),
+            ),
+            (
+                # only one to fail, maybe due to https://github.com/ricardoV94/pymc/blob/mixtures/pymc/distributions/mixture.py#L212
+                # I'm not sure...
+                np.array([0.4, 0.6]),
+                [Normal.dist(-2, 5, size=(11, 7)), Normal.dist(6, 3, size=(11, 7))],
+                (5, 3),
+                np.full(
+                    shape=(
+                        11,
+                        7,
+                        5,
+                        3,
+                    ),
+                    fill_value=2.8,
+                ),
```

Comment:
Not sure if this kind of broadcasting should be working. Can you verify?

Reply:
Yes, we overwrite the size in this case, so I expect the moment shape to be `(5, 3)`. I wonder if we could do something more sensible. You think we should prepend the size? In that case, shouldn't it rather be `(5, 3, 11, 7)`?

Reply:
Thinking again, I don't think we should prepend. The Mixture distribution is still a scalar distribution, so this would be equivalent to trying `pm.Normal(mu=np.zeros((11, 7)), size=(5, 3))` which would fail. The issue here is that we are changing things slightly. We could add `Asserts` but I don't know if the extra complexity is worth it? Maybe a note in the docstrings is sufficient?

Reply:
Broadly speaking, why is size overwritten here? I initially thought that it would get prepended, but this is just me trying to figure things out (i.e. no strong opinions on what should be the case).

Are you suggesting that we should not allow a `size` argument given broadcasted parameters (e.g. `mu=np.zeros((11, 7))`)?

I'm not exactly sure what how we should go about adding `Assert`s as part of the API, but happy to learn about this!

Reply:
An explanation would be pretty involved, maybe we can can have a call over it?

The short answer is that (for convenience) our logp and random methods behave as if the support dimensionality is the same as that of the components. So if you say your Mixture of scalars is of `size=5` the draws should have `shape=5`. To achieve this we resize the components automatically, so the user doesn't have to say each component has a size of `5`.

I have a local branch that allows you to say "treat the components as if they were a single distribution" but it requires more involved logic.

Reply:
You can definitely have vector arguments, but the size must be compatible with these. Just like you cant have a Normal with 3 mus and say it's size is 5. Or a Multivariate Normal with 3 mu vectors and say its size is 5.

This was always the behavior for univariate distributions but up until recently not for multivariates. Aesara use to say size is always prepended. Not anymore. Size must always be equivalent to `shape[:-ndim_supp]`

Reply:
Okay! I think that I get the gist, but yes we can definitely have a chat about this